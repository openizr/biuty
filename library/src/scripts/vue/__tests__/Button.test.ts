/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { mount } from '@vue/test-utils';
import UIButton from 'scripts/vue/Button.vue';

type component = any; // eslint-disable-line @typescript-eslint/no-explicit-any

describe('vue/UIButton', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const wrapper = mount(UIButton, {
      propsData: { label: 'Test', modifiers: 'large' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with id and listeners', async () => {
    const onClick = jest.fn();
    const onFocus = jest.fn();
    const wrapper = mount(UIButton, {
      propsData: {
        id: 'test', label: 'Test', modifiers: 'large',
      },
      listeners: {
        focus: onFocus,
        click: onClick,
      },
    });
    await (wrapper.vm as component).focusButton();
    wrapper.find('button').trigger('click');
    expect(wrapper.html()).toMatchSnapshot();
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onFocus).toHaveBeenCalledTimes(1);
  });

  test('renders correctly - no label', () => {
    const wrapper = mount(UIButton, {
      propsData: {},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - label with icon', () => {
    const wrapper = mount(UIButton, {
      propsData: { label: 'Test', icon: 'star' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - type submit', () => {
    const wrapper = mount(UIButton, {
      propsData: { label: 'Test', type: 'submit' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - icon only', () => {
    const wrapper = mount(UIButton, {
      propsData: { icon: 'star' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - right icon', () => {
    const wrapper = mount(UIButton, {
      propsData: { icon: 'star', iconPosition: 'right' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - disabled', () => {
    const wrapper = mount(UIButton, {
      propsData: { modifiers: 'disabled' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
