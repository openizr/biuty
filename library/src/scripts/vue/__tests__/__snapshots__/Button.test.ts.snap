// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue/UIButton renders correctly - basic 1`] = `
"<button type=\\"button\\" tabindex=\\"0\\" class=\\"ui-button ui-button--large\\">
  <!----> <span class=\\"ui-button__label\\">Test</span>
  <!----></button>"
`;

exports[`vue/UIButton renders correctly - disabled 1`] = `
"<button type=\\"button\\" tabindex=\\"-1\\" class=\\"ui-button ui-button--disabled\\">
  <!---->
  <!---->
  <!----></button>"
`;

exports[`vue/UIButton renders correctly - icon only 1`] = `
"<button type=\\"button\\" tabindex=\\"0\\" class=\\"ui-button ui-button--icon\\"><i class=\\"ui-button__icon\\">star</i>
  <!---->
  <!----></button>"
`;

exports[`vue/UIButton renders correctly - label with icon 1`] = `
"<button type=\\"button\\" tabindex=\\"0\\" class=\\"ui-button\\"><i class=\\"ui-button__icon\\">star</i> <span class=\\"ui-button__label\\">Test</span>
  <!----></button>"
`;

exports[`vue/UIButton renders correctly - no label 1`] = `
"<button type=\\"button\\" tabindex=\\"0\\" class=\\"ui-button\\">
  <!---->
  <!---->
  <!----></button>"
`;

exports[`vue/UIButton renders correctly - right icon 1`] = `
"<button type=\\"button\\" tabindex=\\"0\\" class=\\"ui-button ui-button--icon\\">
  <!---->
  <!----> <i class=\\"ui-button__icon\\">star</i></button>"
`;

exports[`vue/UIButton renders correctly - type submit 1`] = `
"<button type=\\"submit\\" tabindex=\\"0\\" class=\\"ui-button\\">
  <!----> <span class=\\"ui-button__label\\">Test</span>
  <!----></button>"
`;

exports[`vue/UIButton renders correctly - with id and listeners 1`] = `
"<button id=\\"test\\" type=\\"button\\" tabindex=\\"0\\" class=\\"ui-button ui-button--large\\">
  <!----> <span class=\\"ui-button__label\\">Test</span>
  <!----></button>"
`;
